bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/enums/HeatpumpChannel.java

```diff
@@ -0,0 +1,1369 @@
+            HeatpumpVisibility.THERMDESINFEKT),
+    /**
+     * Thermal disinfection (Permanent)
+     * (original: Thermische Desinfektion (Dauerbetrieb))
+     */
+    CHANNEL_EINST_BWTDI_AKT_AL(27, "Einst_BwTDI_akt_AL", SwitchItem.class, null, true,
+            HeatpumpVisibility.THERMDESINFEKT),
+
+    /**
+     * Comfort cooling mode
+     * (original: Comfort Kühlung Betriebsart)
+     */
+    CHANNEL_EINST_BWSTYP_AKT(100, "Einst_BWStyp_akt", NumberItem.class, null, true, HeatpumpVisibility.KUHLUNG),
+
+    /**
+     * Comfort cooling AT release
+     * (original: Comfort Kühlung AT-Freigabe)
+     */
+    CHANNEL_EINST_KUCFTL_AKT(110, "Einst_KuCft1_akt", NumberItem.class, SIUnits.CELSIUS, true,
+            HeatpumpVisibility.KUHLUNG),
+
+    /**
+     * Comfort cooling AT release target
+     * (original: Comfort Kühlung AT-Freigabe Sollwert)
+     */
+    CHANNEL_SOLLWERT_KUCFTL_AKT(132, "Sollwert_KuCft1_akt", NumberItem.class, SIUnits.CELSIUS, true,
+            HeatpumpVisibility.KUHLUNG),
+
+    /**
+     * AT Excess
+     * (original: AT-Überschreitung)
+     */
+    CHANNEL_EINST_KUHL_ZEIT_EIN_AKT(850, "Einst_Kuhl_Zeit_Ein_akt", NumberItem.class, null, true,
+            HeatpumpVisibility.SYSEIN_KUHL_ZEIT_EIN),
+
+    /**
+     * AT undercut
+     * (original: AT-Unterschreitung)
+     */
+    CHANNEL_EINST_KUHL_ZEIT_AUS_AKT(851, "Einst_Kuhl_Zeit_Aus_akt", NumberItem.class, null, true,
+            HeatpumpVisibility.SYSEIN_KUHL_ZEIT_AUS),
+
+    /**
+     * Channel holding complete (localized) status message
+     */
+    CHANNEL_HEATPUMP_STATUS(null, "HauptMenuStatus", StringItem.class, null, false, null);
+
+    private @Nullable Integer channelId;
+    private String command;
+    private Class<? extends Item> itemClass;
+    private @Nullable Unit<?> unit;
+    private Boolean isParameter;
+    private @Nullable HeatpumpVisibility requiredVisibility;
+
+    private HeatpumpChannel(@Nullable Integer channelId, String command, Class<? extends Item> itemClass,
+            @Nullable Unit<?> unit, Boolean isParameter, @Nullable HeatpumpVisibility requiredVisibility) {
+        this.channelId = channelId;
+        this.command = command;
+        this.itemClass = itemClass;
+        this.unit = unit;
+        this.isParameter = isParameter;
+        this.requiredVisibility = requiredVisibility;
+    }
+
+    public @Nullable Integer getChannelId() {
+        return channelId;
+    }
+
+    public String getCommand() {
+        return command;
+    }
+
+    public Class<? extends Item> getItemClass() {
+        return itemClass;
+    }
+
+    public @Nullable Unit<?> getUnit() {
+        return unit;
+    }
+
+    public Boolean isWritable() {
+        return isParameter == Boolean.TRUE;
+    }
+
+    protected @Nullable HeatpumpVisibility getVisibility() {
+        return requiredVisibility;
+    }
+
+    public Boolean isVisible(Integer[] visibilityValues) {
```

Comment:
See above. Use primitives where possible. Please check all.
```suggestion
    public boolean isVisible(Integer[] visibilityValues) {
```